pkg/util/etcdutil/etcdutil.go

```diff
@@ -73,3 +75,23 @@ func MemberNameFromPeerURL(pu string) (string, error) {
 	}
 	return strings.Split(u.Host, ":")[0], nil
 }
+
+func CheckEtcdPodHealth(url string) (bool, error) {
+	cfg := clientv3.Config{
+		Endpoints:   []string{url},
+		DialTimeout: constants.DefaultDialTimeout,
+	}
+	etcdcli, err := clientv3.New(cfg)
+	if err != nil {
+		err := fmt.Errorf("Failed to setup etcd clientv3")
+		return false, err
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), constants.DefaultRequestTimeout)
+	_, err = etcdcli.Get(ctx, "/", clientv3.WithSerializable())
+	cancel()
+	etcdcli.Close()
+	if err != nil {
+		return false, nil
```

Comment:
Well this would not be an error. It just means that the health check failed on this pod.